sql/src/main/java/io/crate/planner/symbol/Function.java

```diff
@@ -100,9 +100,8 @@ public boolean equals(Object o) {
 
         if (arguments != null ? !arguments.equals(function.arguments) : function.arguments != null)
             return false;
-        if (info != null ? !info.equals(function.info) : function.info != null) return false;
+        return !(info != null ? !info.equals(function.info) : function.info != null);
```

Comment:
i think it would make sense to put the info comparison before the arguments comparison, since the argument comparison can be a tree comparison